prime-dotnet-webapi/Services/EmailService.cs

```diff
@@ -304,14 +297,29 @@ public async Task<IEnumerable<string>> GetPharmaNetProvisionerNamesAsync()
                 .ToListAsync();
         }
 
-        private async Task Send(string from, string to, string subject, string body)
+        public async Task<bool> UpdateEmailLogStatuses()
         {
-            await Send(from, new[] { to }, new string[0], subject, body, Enumerable.Empty<(string Filename, byte[] Content)>());
+            var emailLogs = await _context.EmailLogs
+                .Where(e => e.SendType == "CHES"
+                    && e.MsgId != Guid.Empty
+                    && e.LatestStatus != "completed")
+                .ToListAsync();
+
+            foreach (var email in emailLogs)
+            {
+                var status = await _chesClient.GetStatusAsync((Guid)email.MsgId);
```

Comment:
rather than casing, you should be able to just do `email.MsgId` or `email.MsgId.Value`